backend/src/v1/routes/report-routes.ts

```diff
@@ -33,18 +42,74 @@ reportRouter.get(
 
       // get reports by status if status param is provided
       try {
-        const reports = await reportService.getReports(
-          businessGuid,
-          filters,
-        );
+        const reports = await reportService.getReports(businessGuid, filters);
         return res.status(HttpStatus.OK).json(reports);
+      } catch (e) {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).end();
       }
-      catch (e) {
+    },
+  ),
+);
+
+/**
+ * /api/v1/report/:report_id
+ *    accepts 'json':
+ *      Get the input form data for a report in the users business
+ *
+ *    accepts 'html':
+ *      Get an html preview of the report for the users business
+ *
+ *    accepts 'pdf':
+ *      Download pdf of the report for the users business
+ */
+reportRouter.get(
+  '/:report_id',
+  utils.asyncHandler(
+    async (
+      req: Request<{ report_id: string }, null, null, null>,
+      res: Response,
+    ) => {
+      // verifiy business guid
+      const businessGuid =
+        utils.getSessionUser(req)?._json?.bceid_business_guid;
+      if (!businessGuid)
         return res.status(HttpStatus.INTERNAL_SERVER_ERROR).end();
+
+      // params
+      const reportId = req.params.report_id;
+
+      //accepts 'json'
+      if (req.accepts('application/json')) {
+        // get reports by status if status param is provided
+        const report = await reportService.getReportById(
+          businessGuid,
+          reportId,
+        );
+        if (report) return res.status(HttpStatus.OK).json(report);
```

Comment:
What happens if the first if on line 82 is true and the next if at line 88 is false

Reply:
thanks, good eye, each if statement is supposed to either return successfully or fallback to the last line and return an error.